notmuchfs.c

```diff
@@ -1173,9 +1180,37 @@ static int notmuchfs_unlink (const char* path)
  database_close(p_ctx);
 #endif
 
- LOG_TRACE("unlink(%s)\n", path);
- if (unlink(path) != 0)
-   return -errno;
+ if (global_config.delete_tag) {
```

Comment:
Please explicitly compare with != NULL, as is the style used in the rest of this package.